instrumentation/struts/struts-2.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/struts2/ActionInvocationInstrumentation.java

```diff
@@ -28,7 +29,8 @@ public class ActionInvocationInstrumentation implements TypeInstrumentation {
 
   @Override
   public ElementMatcher<ClassLoader> classLoaderOptimization() {
-    return hasClassesNamed("com.opensymphony.xwork2.ActionInvocation");
+    return hasClassesNamed("com.opensymphony.xwork2.ActionInvocation")
+        .and(not(hasClassesNamed("org.apache.struts2.ActionInvocation")));
```

Comment:
I'm not sure why this extra condition is necessary. Since the interface is in different package for struts7 it should clash with the instrumentation for older versions. I'll push to the pr and remove this check if it is needed then please add it back. I'll also move struts 2.3 and 7 instrumentation classes to different packages.

Reply:
Thanks